Zend/tests/gh18519.phpt

```diff
@@ -0,0 +1,39 @@
+--TEST--
+GH-18519: Nested object comparison leading to stack overflow
+--SKIPIF--
+<?php
+if (getenv('SKIP_ASAN')) die('skip as it fatally crash');
+?>
+--FILE--
+<?php
+
+#[AllowDynamicProperties]
+class Node {
+    public $next;
+    // forcing dynamic property creation is key
+}
+
+$first = new Node();
+$first->previous = $first;
+$first->next = $first;
+
+$cur = $first;
+
+for ($i = 0; $i < 50000; $i++) {
```

Comment:
The test "fail" even for i=1 - https://3v4l.org/ak1bc. The comparison probably cannot find cycles correctly.

Reply:
it does trigger the `Nesting level too deep` warning indeed but it does not trigger the `Maximum call stack size reaced` which is the whole point of this test.

Reply:
I have simplified the code and opened https://github.com/php/php-src/issues/18585. I belive the recursion in weak comparasion should be handled.

Reply:
https://github.com/php/php-src/issues/18572#issuecomment-2886434250